### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by rengelke**
# Main changes

### Code Quality Improvements
- Conducted a general code overhaul, including clean-up and formatting
- Migrated error cost functionality within the `ParameterIdentification` class to a new `calculateCostMetrics` function.

### New Features and Functionalities
- Introduced a `calculateCostMetrics` function capable of handling weights, errors, and censored errors
- Fixed the plotting function for cost metrics (`modelCost`)
- Setup of objective function options within `PIConfiguration`

### Testing Enhancements
  - `PIConfiguration`
  - `PiParameters`
  - `PIOutputMappings`
  - `ParameterIdentification`
  - `calculateCostMetrics` function and its helper functions

### CODE REVIEW

File: R/PIConfiguration.R [hunk 1]
```diff
@@ -1,69 +1,73 @@
...
     },
 
-    #' @field targetFunctionType Type of the target function used for error
-    #' calculation. Supported target functions  are listed  in
-    #' `ospsuite.parameteridentification::ObjectiveFunctions`.
-    targetFunctionType = function(value) {
-      if (missing(value)) {
-        private$.targetFunctionType
+    #' @field objectiveFunctionOptions Configuring model fit assessment
+    #' within `.objectiveFunction`. Influences error and model fit calculation.
```
Comment (PavelBal) [inline R/PIConfiguration.R:58]: This documentation is visible to the user, while `.objectiveFunction` is a hidden (private) method. We should not mention any private fields in the user documentation. Just re-phrase it.

File: R/PIConfiguration.R [hunk 2]
```diff
@@ -1,69 +1,73 @@
...
+    #' Supported options and their allowable values are detailed in
+    #' `ospsuite.parameteridentification::ObjectiveFunctionOptions`.
+    objectiveFunctionOptions = function(options = list()) {
+      if (missing(options)) {
+        private$.objectiveFunctionOptions
       } else {
-        validateIsCharacter(value)
-        validateEnumValue(tolower(value), ObjectiveFunctions)
-        private$.targetFunctionType <- value
+        values <- enumWithValues(ObjectiveFunctionOptions)
```
Comment (PavelBal) [inline R/PIConfiguration.R:65]: Why is this required? From what I see, `enumWithValues(ObjectiveFunctionOptions)` returns exactly `ObjectiveFunctionOptions`.

File: R/utilities-helper.R [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+#' @examples
+#' validOptions <- list(
+#'   option1 = c("value1", "value2"),
+#'   option2 = "boolean",
+#'   option3 = "numeric"
+#' )
+#' inputOptions <- list(option1 = "value1", option2 = TRUE, option3 = 5)
+#' validateOptions(inputOptions, validOptions)
+#' @export
+validateIsOption <- function(inputOptions, validOptions) {
```
Comment (PavelBal) [inline R/utilities-helper.R:41]: I think this is a very useful function and belongs to the ospsuite.rutils package. Can you implement it here: https://github.com/Open-Systems-Pharmacology/OSPSuite.RUtils ?

**Review by PavelBal — CHANGES_REQUESTED** [review]

File: R/PIConfiguration.R [hunk 4]
```diff
@@ -110,20 +114,41 @@ PIConfiguration <- R6::R6Class(
       private$.simulateSteadyState <- FALSE
       private$.steadyStateTime <- 1000
       private$.printEvaluationFeedback <- FALSE
-      private$.targetFunctionType <- "lsq"
+      private$.objectiveFunctionOptions <- list(
```
Comment (PavelBal) [inline R/PIConfiguration.R:117]: I hope this is described somewhere :D

**Comment by PavelBal** [comment]
Note: ALWAYS run styler `styler:::style_active_pkg()` before making a PR.

File: R/PIConfiguration.R (hunk 2 above)
Comment (rengelke) [reply R/PIConfiguration.R:65]: I realize it was too much and simplifying will make it more clean.

File: R/PIConfiguration.R [hunk 5]
```diff
@@ -1,69 +1,73 @@
...
+    #' `ospsuite.parameteridentification::ObjectiveFunctionOptions`.
+    objectiveFunctionOptions = function(options = list()) {
+      if (missing(options)) {
+        private$.objectiveFunctionOptions
       } else {
-        validateIsCharacter(value)
-        validateEnumValue(tolower(value), ObjectiveFunctions)
-        private$.targetFunctionType <- value
+        values <- enumWithValues(ObjectiveFunctionOptions)
+        ospsuite.utils::validateIsIncluded(names(options), names(values))
```
[the remark this answers is not in the document]
Comment (rengelke) [reply R/PIConfiguration.R:66]: This part will be changed as mentioned above

File: R/PIConfiguration.R [hunk 6]
```diff
@@ -1,69 +1,73 @@
...
+        validateIsOption(options, values)
```
[the remark this answers is not in the document]
Comment (rengelke) [reply R/PIConfiguration.R:67]: Reviewed the function and implemented it. Will use it from ospsuite.utils after

File: R/PIConfiguration.R (hunk 1 above)
Comment (rengelke) [reply R/PIConfiguration.R:58]: I see your point. Fixed.

File: R/PIConfiguration.R (hunk 4 above)
Comment (rengelke) [reply R/PIConfiguration.R:116]: Description still needed. I guess it would be a good to have it somewhere together with the description of the optimization algorithm options

File: R/enums.R [hunk 7]
```diff
@@ -39,7 +39,42 @@ AlgorithmOptions <- enum(c(
   "steptol" # step count before checking relative tolerance
 ))
 
-#' @title ObjectiveFunctions
-#' List of supported objective functions to calculate the error.
+#' @title ObjectiveFunctionOptions
+#' List of supported parameter options for calculation of model cost.
 #' @export
-ObjectiveFunctions <- enum(c("lsq", "m3"))
+ObjectiveFunctionOptions <- list(
```
Comment (PavelBal) [inline R/enums.R:45]: This is the place to describe the options. And later in the vignette.

**Review by PavelBal — APPROVED** [review]
